### ISSUE TITLE
IMAX b6 Mini ver 1.13 doesn't work

### ISSUE THREAD

**Issue by alexwlpr** [Milek7/imax-b6mini-datalogger#2]
Hello. This script doesn't work on version 1.13. But charge master work correctly. I check length of data after:
data = h.read(64, 1000)
it returns 0.
I think in this version changed request ([0x0F, 0x03, 0x55, 0x00, 0x55, 0xFF, 0xFF, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00]), because LogView Studio send same request and doesn't work too.
I never work with HID devices and don't know how analyze all traffic between PC and HID device.
Sorry for my english, i'm from Russia.

**Comment by alexwlpr** [comment]
Yes!
I find in internet HID sniffer, and sniff charge master. Correct data to send:
h.write([0x00, 0x0F, 0x03, 0x55, 0x00, 0x55, 0xFF, 0xFF, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00])
and this array need always send to device (instead h.write([0]))
I receive correct data for energy, timer, voltage, current, int temp
I don't have temp sensor, and test on 1 cell accumulator. But in cells i receive strange data:
0.281, 0.28, 0.3, 0.296, 0.31, 0.31 and this data changes in a little.

**Comment by Milek7** [comment]
Yeah, that h.write([0]) was a hack, previous firmware didn't care what was send and always sent status packet in reply.
Are you sure that you connected battery? These ~0.3V values looks like unconnected cells.

**Comment by alexwlpr** [comment]
> Are you sure that you connected battery? These ~0.3V values looks like unconnected cells.

No, balance not connected (1 cell, 18650 battery). I also think that it is not connected cells.
LogoView studio send correct data, but can't receive data.

**Comment by Milek7** [comment]
Ok, if balance port is not connected then you only get voltage readings in 4th column.
I can't help you with LogView, never used it.

**Comment by alexwlpr** [comment]
> I can't help you with LogView, never used it.

I'm understand it. I bought imax today and learn its capabilities.

**Comment by beikeland** [comment]
Have you guys had any luck initiating resistance measurement from USB?

**Comment by Milek7** [comment]
No, I haven't tried that. IIRC ChargeMaster software doesn't have this feature, so I don't think it is possible with original firmware.
